### PR TITLE
CDAP-3637 removing ApplicationTemplate and Adapter from api

### PR DESCRIPTION
**Pull request by albertshau**
This change removes ApplicationTemplate and Adapter related
classes from the cdap api. It also cleans up a lot of the code
that became unused due to the removal of the classes. This results
in a LOT of code removed.

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/DynamicMapReduceContext.java [hunk 1]
```diff
@@ -82,12 +81,10 @@ public DynamicMapReduceContext(Program program,
                                  TransactionSystemClient txClient,
                                  DatasetFramework dsFramework,
                                  LocationFactory locationFactory,
-                                 @Nullable AdapterDefinition adapterSpec,
-                                 @Nullable PluginInstantiator pluginInstantiator,
                                  @Nullable PluginInstantiator artifactPluginInstantiator) {
     super(program, type, runId, taskId, runtimeArguments, Collections.<String>emptySet(), spec,
           logicalStartTime, programNameInWorkflow, workflowToken, discoveryServiceClient, metricsCollectionService,
-          dsFramework, locationFactory, adapterSpec, pluginInstantiator, artifactPluginInstantiator);
+          dsFramework, locationFactory, artifactPluginInstantiator);
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/DynamicMapReduceContext.java]: can be renamed to pluginInstantiator?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceClassLoader.java [hunk 2]
```diff
@@ -101,11 +98,8 @@ public MapReduceClassLoader(ClassLoader programClassLoader) {
   public MapReduceClassLoader(ClassLoader programClassLoader,
                               Configuration hConf,
                               Map<String, Plugin> plugins,
-                              @Nullable AdapterDefinition adapterSpec,
-                              @Nullable PluginInstantiator pluginInstantiator,
                               @Nullable PluginInstantiator artifactPluginInstantiator) {
-    this(new Parameters(programClassLoader, hConf, plugins, adapterSpec,
-                        pluginInstantiator, artifactPluginInstantiator));
+    this(new Parameters(programClassLoader, hConf, plugins, artifactPluginInstantiator));
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceClassLoader.java]: same here

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/DynamicMapReduceContext.java (hunk 1 above)
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/DynamicMapReduceContext.java]: or was it to reduce conflicts?

Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/DynamicMapReduceContext.java]: i guess we can make that change in another PR later

**Comment by gokulavasan** [comment]
LGTM :+1:

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/DynamicMapReduceContext.java (hunk 1 above)
Comment (albertshau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/DynamicMapReduceContext.java]: renamed in some places, but missed this one.  Will rename it

**Pull request merged**